pycytominer/cyto_utils/cells.py

```diff
@@ -469,26 +502,35 @@ def aggregate_compartment(
             )
 
             if compute_counts and self.fields_of_view_feature not in self.strata:
-                fields_count_df = self.image_df.loc[
-                    :, list(np.union1d(self.strata, self.fields_of_view_feature))
-                ]
-                fields_count_df = (
-                    fields_count_df.groupby(self.strata)[self.fields_of_view_feature]
-                    .count()
-                    .reset_index()
-                    .rename(
-                        columns={
-                            f"{self.fields_of_view_feature}": f"Metadata_Site_Count"
-                        }
-                    )
+                fields_count_df = aggregate_fields_count(
+                    self.image_df, self.strata, self.fields_of_view_feature
                 )
 
+                if add_image_features:
+                    fields_count_df = aggregate_image_features(
+                        fields_count_df,
+                        self.image_features_df,
+                        self.image_feature_categories,
+                        self.image_cols,
+                        self.strata,
+                        self.aggregation_operation,
+                    )
+
                 partial_object_df = fields_count_df.merge(
                     partial_object_df,
                     on=self.strata,
                     how="right",
                 )
 
+                # Separate all the metadata and feature columns.
+                metadata_cols = infer_cp_features(partial_object_df, metadata=True)
+                feature_cols = infer_cp_features(partial_object_df, image_features=True)
+
+                partial_object_df = pd.concat(
+                    [partial_object_df[metadata_cols], partial_object_df[feature_cols]],
+                    axis=1,
+                )
```

Comment:
All you're doing here is reordering the columns? I think you can use [`reindex`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.reindex.html):

```python
# Something like
partial_object_df = partial_object_df.reindex(columns=metadata_cols + feature_cols)
```